app/src/main/java/com/fish/monsters/features/settings/ui/SettingsScreen.kt

```diff
@@ -0,0 +1,138 @@
+package com.fish.monsters.features.settings.ui
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.fish.monsters.R
+import com.fish.monsters.common.models.ui.Language
+import com.fish.monsters.common.utils.settings.SettingsGlobalState
+import com.fish.monsters.common.views.CapText
+import com.fish.monsters.common.views.FadingHorizontalDivider
+import com.fish.monsters.common.views.FishSwitch
+import com.fish.monsters.common.views.PreviewContainer
+import com.fish.monsters.common.views.buttons.DangerOutlinedFishButton
+import com.fish.monsters.common.views.buttons.DropdownTextButton
+import com.fish.monsters.common.views.screenContent.ScreenBox
+import com.fish.monsters.core.SettingsUris
+import com.fish.monsters.features.settings.SettingsViewModel
+import com.fish.monsters.features.settings.ui.components.SettingsButtonGridSection
+import com.fish.monsters.features.settings.ui.components.SettingsClearProgressDialog
+import com.fish.monsters.features.settings.ui.components.SettingsRowSection
+import com.fish.monsters.features.settings.ui.components.SettingsSliderSection
+import org.koin.compose.koinInject
+
+
+@Composable
+fun SettingsScreen(viewModel: SettingsViewModel = koinInject()) {
+    val isClearProgressDialogVisible = remember { mutableStateOf(false) }
+    SettingsScreenContent(
+        viewModel.settingsGlobalState.value, isClearProgressDialogVisible
+    ) { viewModel.invokeEvent(it) }
+}
+
+@Composable
+private fun SettingsScreenContent(
+    settingsGlobalState: SettingsGlobalState,
+    isClearProgressDialogVisible: MutableState<Boolean>,
+    invokeEvent: (event: SettingsEvent) -> Unit
```

Comment:
Tutaj zabieg opisany powyżej, do każdej funkcji view model mamy dostęp poprzez funkcje `invokeEvent()` którą tu podsyłamy, dodatkowo osobno podajemy state, dzięki temu możemy łatwo to potem testować a także w preview wyświetlić ekran dokładnie w takim stanie jaki nam się podoba.